testutils/src/main/scala/za/co/absa/enceladus/testutils/datasetComparison/ComparisonJob.scala

```diff
@@ -53,11 +76,17 @@ object ComparisonJob {
     val actualMinusExpected: Dataset[Row] = actualDf.except(expectedDf)
 
     if (expectedMinusActual.count() != 0 || actualMinusExpected.count() != 0) {
-      expectedMinusActual.write.format("parquet").save(s"${cmd.outPath}/expected_minus_actual")
-      actualMinusExpected.write.format("parquet").save(s"${cmd.outPath}/actual_minus_expected")
+      if (cmd.keys.isDefined) {
+        val joinedData = getKeyBasedOutput(expectedMinusActual, actualMinusExpected, cmd.keys.get)
```

Comment:
Already discussed, but just in case ) The datasets are not generally expected to have row duplicates. However, it may happen when there are aggregated datasets for multiple dates, or if there are bugs in the source system. So it would be useful to have a fail safe check at the beginning (before joining): e.g., group by keys each dataset and indicate (or maybe throw exception) if count > 1

Reply:
It is done. But it works only if one or more keys (unique ids) are specified.

Reply:
Will wait for your approval.